### PR TITLE
Update Noah-MP MPTABLE to WRFv4.1

### PR DESCRIPTION
**Pull request by barlage**
TYPE: version consolidation

KEYWORDS: Noah-MP

SOURCE: Michael Barlage (NCAR)

DESCRIPTION OF CHANGES:

Add Noah-MP MPTABLE changes to unify community code with WRF v4.1.

Produces no answer changes.

LIST OF MODIFIED FILES:
M module_sf_noahmplsm.F
M module_sf_noahmpdrv.F
M MPTABLE.TBL

TESTS CONDUCTED:

1. One-year simulation (Cheyenne, PGI) in two calibration basins produces no changes

### CODE REVIEW

File: trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F [hunk 1]
```diff
@@ -9268,12 +9279,18 @@ subroutine read_mp_veg_parameters(DATASET_IDENTIFIER)
     ISICE_TABLE        = -99999
     ISCROP_TABLE       = -99999
     EBLFOREST_TABLE    = -99999
+    NATURAL_TABLE      = -99999
+    LOW_DENSITY_RESIDENTIAL_TABLE   = -99999
+    HIGH_DENSITY_RESIDENTIAL_TABLE  = -99999
+    HIGH_INTENSITY_INDUSTRIAL_TABLE = -99999
+
+    inquire( file='MPTABLE.TBL', exist=file_named )
```
Comment (jmccreight) [inline trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F]: Acceptable on WCOSS?
Works on WCOSS?
How to test on WCOSS?

Comment (jmccreight) [reply trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F]: @donaldwj 
I think we should just accept this and deal with the issue later, when the time comes.

File: trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F [hunk 2]
```diff
@@ -9411,11 +9433,13 @@ subroutine read_mp_soil_parameters()
 !
 !-----READ IN SOIL PROPERTIES FROM SOILPARM.TBL
 !
-#ifndef NCEP_WCOSS
-    OPEN(21, FILE='SOILPARM.TBL',FORM='FORMATTED',STATUS='OLD',IOSTAT=ierr)
-#else
-    OPEN(21, FORM='FORMATTED',STATUS='OLD',IOSTAT=ierr)
-#endif
+    inquire( file='SOILPARM.TBL', exist=file_named )
```
Comment (jmccreight) [inline trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F]: same pattern here.

File: trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F [hunk 3]
```diff
@@ -9441,11 +9465,13 @@ subroutine read_mp_soil_parameters()
 !
 !-----READ IN GENERAL PARAMETERS FROM GENPARM.TBL
 !
-#ifndef NCEP_WCOSS
-    OPEN(22, FILE='GENPARM.TBL',FORM='FORMATTED',STATUS='OLD',IOSTAT=ierr)
-#else
-    OPEN(22, FORM='FORMATTED',STATUS='OLD',IOSTAT=ierr)
-#endif
+    inquire( file='GENPARM.TBL', exist=file_named )
```
Comment (jmccreight) [inline trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F]: and here

File: trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F (hunk 2 above)
Comment (jmccreight) [reply trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F]: there are 2 more of these for a total of 5.

File: trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F (hunk 1 above)
Comment (donaldwj) [reply trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F]: If NCO has an issue with this they will say something when we try to hand off the code, I haven't had a chance to determine if this violates any of their rules.

Comment (jmccreight) [reply trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F]: I will take that as a "go ahead and merge  and we will deal with issues on WCOSS later".

Comment (rcabell) [reply trunk/NDHMS/Land_models/NoahMP/phys/module_sf_noahmplsm.F]: Would this ever return true on WCOSS? If it's not opened by name, it shouldn't, right? I think the way it's handled on WCOSS the actual on-disk filename will be something akin to "`fort.15`" and therefore it should be ok. I'm alright with the shoot-first, ask-questions-later approach with WCOSS for now.

**Pull request merged**